### DISCUSSION TITLE
How to upload files through the api docs?

### DISCUSSION THREAD

**Discussion by spa5k** [danielgtaylor/huma#397] [Q&A] [answered]
I basically want a file upload button as we can do in the swagger ui by doing this 

```go
@Param image formData file true "Asset file to upload"
```

here is my sample route - 

```go
	type CreateReviewInput struct {
		Comment           string                  `db:"comment" json:"comment"`
		AttachedFileNames []string                `db:"attached_file_names" json:"attached_file_names"`
	}

	type CreateReviewRequest struct {
		UserAgent string            `header:"User-Agent"`
		Body      CreateReviewInput `json:"body"`
		RawBody   multipart.Form
	}

	huma.Register(api, huma.Operation{
		Path:        path,
		Method:      "POST",
		Tags:        []string{"Review"},
		OperationID: "create-review",
		Middlewares: huma.Middlewares{middleware.UserAuthenticationMiddleware(tokenService)},
	}, func(ctx context.Context, i *CreateReviewRequest) (*createReviewResponse, error) {}
	
}
```

I want it to appear like this

![image](https://github.com/danielgtaylor/huma/assets/79936503/03c34d62-a9c6-4c0a-ae46-da46bec82bb1)

Apparently this needs to be setup  https://docs.stoplight.io/docs/platform/c2f20944173d7-work-with-media-types

```
"requestBody": {
       "content": {
         "multipart/form-data": {
           "schema": {
             "type": "object",
             "properties": {
               "file": {
                 "type": "string",
                 "format": "binary"
               }
             }
           }
         }
         ```

**Comment by spa5k** [comment]
Currently, the UI comes like this - 

![image](https://github.com/danielgtaylor/huma/assets/79936503/0940b53e-eed0-42ba-a6b6-ec662357d0db)

**Comment by spa5k** [comment]
Hey @danielgtaylor got any idea on this? If this needs some contribution, do tell me. Thanks.

**Reply by danielgtaylor** [reply]
Yeah this seems like it needs a bit of work to get the schema right for the various docs tools to render the upload button properly. If you want to create a PR go for it!

**Comment by lsdch** [comment, marked as answer]
Stoplight wants either `contentEncoding: "base64"` or `contentMediaType: "application/octet-stream"` [to identify a field as a file input](https://github.com/stoplightio/elements/blob/d7499d7caadf46771bcdae98eff9a38feca7c708/packages/elements-core/src/components/TryIt/Parameters/parameter-utils.ts#L33). 

#422 aims to fix that by adding `contentMediaType: "application/octet-stream"` whenever the format defined is the schema is `binary`. 
Alternatively, you could use base64 (`schema.ContentEncoding = "base64"`) encoding if that works for you, though that involves encoding/decoding the b64 string both on the client and server sides.

On a side note, draft PR #415 is on the way to improve support and documentation for file uploads.
